lib/v2/screens/profile_screens/profile/interactor/usecases/save_image_use_case.dart

```diff
@@ -0,0 +1,20 @@
+import 'dart:io';
+
+import 'package:async/async.dart';
+import 'package:meta/meta.dart';
+import 'package:firebase_storage/firebase_storage.dart';
+import 'package:path/path.dart' as path_lib;
+import 'package:seeds/v2/datasource/local/settings_storage.dart';
+import 'package:uuid/uuid.dart';
+
+class SaveImageUseCase {
+  Future<Result> run({@required File file}) async {
+    String extensionName = path_lib.extension(file.path);
+    String path = "ProfileImage/" + settingsStorage.accountName + '/' + Uuid().v4() + extensionName;
+    Reference reference = FirebaseStorage.instance.ref().child(path);
+    String fileType = extensionName.isNotEmpty ? extensionName.substring(1) : '*';
+    await reference.putFile(file, SettableMetadata(contentType: "image/$fileType"));
+    var url = await reference.getDownloadURL();
+    return ValueResult(url);
+  }
+}
```

Comment:
@gguijarro-c-chwy Actually I'm not sure how to handle this fire store logic I put it as a use case but I'm not sure what approach to use to catch exceptions for fire store

Reply:
this is fine for now, Ideally this is moved to a service that handles IO
and the service can return ValueResult or ErrorResult

Then this use case calls that service etc.